dog.md

```diff
@@ -0,0 +1,208 @@
+         <dog-ilks> ... ILK |-> Ilk( ... hole: (_ => HOLE) ) ... </dog-ilks>
+
+    rule <k> Dog . file chop ILK CHOP => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... chop: (_ => CHOP) ) ... </dog-ilks>
+      requires CHOP >=Wad wad(1)
+
+    rule <k> Dog . file clip ILK CLIP => . ... </k>
+         <dog-ilks> ... ILK |-> Ilk( ... clip: (_ => CLIP) ) ... </dog-ilks>
+         <clip-ilk> CLIP_ILK </clip-ilk>
+      requires ILK ==K CLIP_ILK
+```
+
+Dog Semantics
+-------------
+
+```k
+    syntax DogStep ::= "bark" String Address Address
+ // ------------------------------------------------
+    rule <k> Dog . bark ILK URN KPR
+    =>  #let ROOM = minRad( ( DOG_HOLE -Rad DOG_DIRT), (HOLE -Rad  DIRT) ) #in (
+        #let DART_INITIAL = minWad( ART, ( ( ROOM /Rate RATE) /Wad CHOP) ) #in (
+        #let DART_FINAL = (#if (ART >Wad DART_INITIAL) andBool ( ((ART -Wad DART_INITIAL) *Rate RATE) <Rad DUST ) #then ART #else   DART_INITIAL #fi) #in (
+        #let NON_DUSTY = (#if (ART >Wad DART_INITIAL) andBool ( ((ART -Wad DART_INITIAL) *Rate RATE) <Rad DUST ) #then true #else (DART_INITIAL *Rate RATE) >=Rad DUST  #fi) #in (
+        #let DINK = (INK *Wad DART_FINAL) /Wad ART #in (
+        #let DUE = DART_FINAL *Rate RATE #in (
+        #let TAB = DUE *RadWad2Rad CHOP #in (
+    NON_DUSTY
+    ~> KPR
+    ~> TAB
+    ~> emitBark ILK URN DINK DART_FINAL DUE CLIP 0 ) ) ) ) ) ) )
+    ... </k>
+         <dog-live> DOG_LIVE  </dog-live>
+         <dog-dirt> DOG_DIRT  </dog-dirt>
+         <dog-hole> DOG_HOLE  </dog-hole>
+         <vat-urns> ... { ILK, URN } |-> Urn( INK, ART ) ...                                                </vat-urns>
+         <dog-ilks> ... ILK |-> Ilk( ... clip: CLIP:ClipContract, chop: CHOP, hole: HOLE,  dirt: DIRT ) ... </dog-ilks>
+         <vat-ilks> ... ILK |-> Ilk( ... rate: RATE, spot: SPOT, dust: DUST ) ...                           </vat-ilks>
+      requires DOG_LIVE ==Bool true
+      andBool ( ( SPOT >Ray ray(0) )
+      andBool ( (INK *Rate SPOT ) <Rad (ART *Rate RATE) ) )
+      andBool ( (DOG_HOLE >Rad DOG_DIRT )
+      andBool ( HOLE >Rad DIRT ) )
+
+    rule <k> NON_DUSTY:Bool ~> KPR:Address ~> TAB:Rad ~> emitBark ILK URN DINK DART_FINAL DUE CLIP _
+    => call DOG_VAT . grab ILK URN CLIP DOG_VOW (wad(0) -Wad DINK) (wad(0) -Wad DART_FINAL)
+    ~> call DOG_VOW . fess DUE
+    ~> call CLIP . kick TAB DINK URN KPR
+    ~> emitBark ILK URN DINK DART_FINAL DUE CLIP 0
+    ... </k>
+         <dog-vow>  DOG_VOW   </dog-vow>
+         <dog-vat>  DOG_VAT   </dog-vat>
+         <dog-ilks> ... ILK |-> Ilk( ... clip: CLIP:ClipContract ) ... </dog-ilks>
+         <vat-urns> ... { ILK, URN } |-> Urn( _, ART) ...                     </vat-urns>
+         <vat-ilks> ... ILK |-> Ilk( ... rate: RATE, spot: _, dust: DUST) ... </vat-ilks>
+      requires (#if (ART >Wad DART_FINAL) #then (#if ( ( ART -Wad DART_FINAL ) *Rate RATE ) <Rad DUST #then true #else ( ( DART_FINAL *Rate RATE ) >=Rad DUST ) #fi ) #else true #fi)
```

Comment:
This condition doesn't make sense to me. Let's take just the `true` branch of the first conditional:

```
#if (ART >Wad DART_FINAL) #then (#if ( ( ART -Wad DART_FINAL ) *Rate RATE ) <Rad DUST #then true #else ( ( DART_FINAL *Rate RATE ) >=Rad DUST ) #fi )
```

That's saying "if we did a partial liquidation, then if the remnant was was dusty, evaluate to `true`, else return the truth value of whether the auction itself is non-dusty". But we never leave a dusty remnant in the contract code--it always switches to a full a liquidation. And this logic seems to be encoded correctly above when determining `DART_FINAL`. So we only ever take one branch of that conditional. I'm not 100% sure but it looks like you're maybe just trying to re-express `NON_DUSTY` here and can drop the condition? Or am I missing what this actually captures from the code?

Reply:
I am in fact, re-expressing `NON_DUSTY`, but incorrectly by using the final values of the mentioned variables, I'm removing this condition

Reply:
Thanks!